src/Maestro/FeedCleaner/FeedCleaner.cs

```diff
@@ -0,0 +1,369 @@
+        /// Get a mapping of feed -> (package, versions) for the release feeds so it
+        /// can be easily queried whether a version of a package is in a feed.
+        /// </summary>
+        /// <returns>Mapping of packages to versions for the release feeds.</returns>
+        private async Task<Dictionary<string, Dictionary<string, HashSet<string>>>> GetPackagesForReleaseFeedsAsync()
+        {
+            var packagesWithVersionsInReleaseFeeds = new Dictionary<string, Dictionary<string, HashSet<string>>>();
+            IEnumerable<(string account, string project, string feedName)> dotnetManagedFeeds = Options.ReleasePackageFeeds;
+            foreach ((string account, string project, string feedName) in dotnetManagedFeeds)
+            {
+                string readableFeedURL = ComputeAzureArtifactsNuGetFeedUrl(feedName, account, project);
+
+                packagesWithVersionsInReleaseFeeds[readableFeedURL] = await GetPackageVersionsForFeed(account, project, feedName);
+            }
+            return packagesWithVersionsInReleaseFeeds;
+        }
+
+        /// <summary>
+        /// Construct a nuget feed URL for an Azure DevOps Artifact feed
+        /// </summary>
+        /// <param name="feedName">Name of the feed</param>
+        /// <param name="account">Azure DevOps Account where the feed is hosted</param>
+        /// <param name="project">Optional project for the feed.</param>
+        /// <returns>Url of the form https://pkgs.dev.azure.com/account/project/_packaging/feedName/nuget/v3/index.json </returns>
+        private static string ComputeAzureArtifactsNuGetFeedUrl(string feedName, string account, string project = null)
+        {
+            string projectSection = string.IsNullOrEmpty(project) ? "" : $"{project}/";
+            return $"https://pkgs.dev.azure.com/{account}/{projectSection}_packaging/{feedName}/nuget/v3/index.json";
+        }
+
+        /// <summary>
+        /// Gets a Mapping of package -> versions for an Azure DevOps feed.
+        /// </summary>
+        /// <param name="azdoClient">Azure DevOps client.</param>
+        /// <param name="account">Azure DevOps account.</param>
+        /// <param name="project">Azure DevOps project the feed is hosted in.</param>
+        /// <param name="feedName">Name of the feed</param>
+        /// <returns>Dictionary where the key is the package name, and the value is a HashSet of the versions of the package in the feed</returns>
+        private async Task<Dictionary<string, HashSet<string>>> GetPackageVersionsForFeed(string account, string project, string feedName)
+        {
+            var azdoClient = AzureDevOpsClients[account];
+            var packagesWithVersions = new Dictionary<string, HashSet<string>>();
+            List<AzureDevOpsPackage> packagesInFeed = await azdoClient.GetPackagesForFeedAsync(account, project, feedName);
+            foreach (AzureDevOpsPackage package in packagesInFeed)
+            {
+                {
```

Comment:
Needless `{ }`?

Reply:
Probably left over from another version where I had another loop in there. Will remove.